File: Presentations/WG2_11/GlassBR_Code/fromGOOL/csharp/OutputFormat.cs
using System;
using System.IO;
using System.Collections;
using System.Collections.Generic;

namespace GlassBR {
    public class OutputFormat {
        
        public static void display_output(string filename, double q, double j, double q_hat_tol, double pb, double lr, double nfl, Boolean is_safe1, Boolean is_safe2, InputParameters inparams) {
            StreamWriter outfile;
            outfile = new StreamWriter(filename);
            outfile.Write("a           ");
            outfile.WriteLine(inparams.a);
            outfile.Write("b           ");
            outfile.WriteLine(inparams.b);
            outfile.Write("t           ");
            outfile.WriteLine(inparams.t);
            outfile.Write("w           ");
            outfile.WriteLine(inparams.w);
            outfile.Write("tnt         ");
            outfile.WriteLine(inparams.tnt);
            outfile.Write("sdx         ");
            outfile.WriteLine(inparams.sdx);
            outfile.Write("sdy         ");
            outfile.WriteLine(inparams.sdy);
            outfile.Write("sdz         ");
            outfile.WriteLine(inparams.sdz);
            outfile.Write("pbtol       ");
            outfile.WriteLine(inparams.pbtol);
            outfile.Write("asprat      ");
            outfile.WriteLine(inparams.asprat);
            outfile.Write("sd          ");
            outfile.WriteLine(inparams.sd);
            outfile.Write("h           ");
            outfile.WriteLine(inparams.h);
            outfile.Write("gtf         ");
            outfile.WriteLine(inparams.gtf);
            outfile.Write("ldf         ");
            outfile.WriteLine(inparams.ldf);
            outfile.Write("wtnt        ");
            outfile.WriteLine(inparams.wtnt);
            outfile.Write("E           ");
            outfile.WriteLine(inparams.E);
            outfile.Write("td          ");
            outfile.WriteLine(inparams.td);
            outfile.Write("m           ");
            outfile.WriteLine(inparams.m);
            outfile.Write("k           ");
            outfile.WriteLine(inparams.k);
            outfile.Write("lsf         ");
            outfile.WriteLine(inparams.lsf);
            outfile.Write("gt          ");
            outfile.WriteLine(inparams.gt);
            outfile.Write("Demand (q)                      ");
            outfile.WriteLine(q);
            outfile.Write("Stress Distr. Factor (j)        ");
            outfile.WriteLine(j);
            outfile.Write("Tolerable Pressure (q_hat_tol)  ");
            outfile.WriteLine(q_hat_tol);
            outfile.Write("Prob. of Breakage (pb)          ");
            outfile.WriteLine(pb);
            outfile.Write("Capacity (lr)                   ");
            outfile.WriteLine(lr);
            outfile.Write("Non-Factored Load (nfl)         ");
            outfile.WriteLine(nfl);
            outfile.Write("Safety Req. 1 (is_safe1)        ");
            outfile.WriteLine(is_safe1);
            outfile.Write("Safety Req. 2 (is_safe2)        ");
            outfile.WriteLine(is_safe2);
            if (is_safe1 && is_safe2) {
                outfile.WriteLine("For the given input parameters, the glass is considered safe.");
            }
            else {
                outfile.WriteLine("For the given input parameters, the glass is NOT considered safe.");
            }
            outfile.Close();
        }
    }
}

File: Presentations/WG2_11/GlassBR_Code/fromGOOL/java/OutputFormat.java
package GlassBR;

import java.util.Arrays;
import java.util.BitSet;
import java.util.Scanner;
import java.io.PrintWriter;
import java.io.File;
import java.util.Vector;

public class OutputFormat {
    
    public static void display_output(String filename, double q, double j, double q_hat_tol, double pb, double lr, double nfl, Boolean is_safe1, Boolean is_safe2, InputParameters inparams) throws Exception {
        PrintWriter outfile;
        outfile = new PrintWriter(filename);
        outfile.print("a           ");
        outfile.println(inparams.a);
        outfile.print("b           ");
        outfile.println(inparams.b);
        outfile.print("t           ");
        outfile.println(inparams.t);
        outfile.print("w           ");
        outfile.println(inparams.w);
        outfile.print("tnt         ");
        outfile.println(inparams.tnt);
        outfile.print("sdx         ");
        outfile.println(inparams.sdx);
        outfile.print("sdy         ");
        outfile.println(inparams.sdy);
        outfile.print("sdz         ");
        outfile.println(inparams.sdz);
        outfile.print("pbtol       ");
        outfile.println(inparams.pbtol);
        outfile.print("asprat      ");
        outfile.println(inparams.asprat);
        outfile.print("sd          ");
        outfile.println(inparams.sd);
        outfile.print("h           ");
        outfile.println(inparams.h);
        outfile.print("gtf         ");
        outfile.println(inparams.gtf);
        outfile.print("ldf         ");
        outfile.println(inparams.ldf);
        outfile.print("wtnt        ");
        outfile.println(inparams.wtnt);
        outfile.print("E           ");
        outfile.println(inparams.E);
        outfile.print("td          ");
        outfile.println(inparams.td);
        outfile.print("m           ");
        outfile.println(inparams.m);
        outfile.print("k           ");
        outfile.println(inparams.k);
        outfile.print("lsf         ");
        outfile.println(inparams.lsf);
        outfile.print("gt          ");
        outfile.println(inparams.gt);
        outfile.print("Demand (q)                      ");
        outfile.println(q);
        outfile.print("Stress Distr. Factor (j)        ");
        outfile.println(j);
        outfile.print("Tolerable Pressure (q_hat_tol)  ");
        outfile.println(q_hat_tol);
        outfile.print("Prob. of Breakage (pb)          ");
        outfile.println(pb);
        outfile.print("Capacity (lr)                   ");
        outfile.println(lr);
        outfile.print("Non-Factored Load (nfl)         ");
        outfile.println(nfl);
        outfile.print("Safety Req. 1 (is_safe1)        ");
        outfile.println(is_safe1);
        outfile.print("Safety Req. 2 (is_safe2)        ");
        outfile.println(is_safe2);
        if (is_safe1 && is_safe2) {
            outfile.println("For the given input parameters, the glass is considered safe.");
        }
        else {
            outfile.println("For the given input parameters, the glass is NOT considered safe.");
        }
        outfile.close();
    }
}

